src/inference/src/dev/converter_utils.cpp

```diff
@@ -273,7 +273,7 @@ class IInferencePluginWrapper : public InferenceEngine::IInferencePlugin {
     }
 
     void SetCore(std::weak_ptr<InferenceEngine::ICore> core) override {
-        return m_plugin->set_core(std::dynamic_pointer_cast<ov::ICore>(core));
+        return m_plugin->set_core(ov::Any(core).as<std::shared_ptr<ov::ICore>>());
```

Comment:
Why change it?

Reply:
> Why change it?

@riverlijunjie  std::dynamic_pointer_cast used here is not the standards func from std, which is is the api (deprecated) defined in Any.